src/Compilers/CSharp/Portable/BoundTree/Constructors.cs

```diff
@@ -60,6 +60,7 @@ private static bool NeedsByValueFieldAccess(BoundExpression? receiver, FieldSymb
         {
             if (fieldSymbol.IsStatic ||
                 !fieldSymbol.ContainingType.IsValueType ||
+                fieldSymbol.RefKind is RefKind.Ref ||
```

Comment:
In L76 we are effectively checking `RefKind != RefKind.None` on the `localSymbol`, should we also do that here for the `fieldSymbol`? Is it possible we need this when we have a ref-readonly field whose referent value is changing "during" the expression?

Reply:
If we have a `ref readonly` field, there are binding errors if we try to change its referent value, so I think we cannot observe the difference (the `IsByValue` computed here is used by `Binder.HasHome` in codegen to determine whether the field should be stored in a temp but we never get there if we have binding errors). But it makes sense to me to check what you suggest.